src/NServiceBus.Core/Serializers/XML/XmlDeserialization.cs

```diff
@@ -309,7 +309,13 @@ object GetObjectOfTypeFromNode(Type t, XmlNode node)
                     type = Type.GetType($"System.{n.Name.Substring(0, n.Name.IndexOf(":"))}", false, true);
                 }
 
-                var prop = GetProperty(t, n.Name);
+                if (!cache.typeMembers.TryGetValue(t, out var typeMembers))
+                {
+                    cache.InitType(t);
+                    cache.typeMembers.TryGetValue(t, out typeMembers);
```

Comment:
Why a `TryGetValue` here and a "get" here https://github.com/Particular/NServiceBus/pull/5123/files#diff-952951b2c63349554d3e4903ddd8ced1R157  ?

Reply:
Ok, I see that now, `WriteObject` never calls `Write` for simple types etc so that's why the get works.

Yea a comment would probably be a good idea.

That said we could change the code to not check for type members for those types and change the code to do a get instead but I'd say that's to risky at this stage?

Reply:
> That said we could change the code to not check for type members for those types and change the code to do a get instead but I'd say that's to risky at this stage?

yeah, I tried to not touch the existing behavior as good as I could. Not something I want to touch at this point.